improver/utilities/mathematical_operations.py

```diff
@@ -356,3 +356,54 @@ def process(self, cube):
         integrated_cube = self.perform_integration(upper_bounds_cube, lower_bounds_cube)
 
         return integrated_cube
+
+
+def alinfit(x, y, axis=-1, gradient_only=False):
+    """Uses a simple linear fit approach to calculate the
+    gradient. Although equivalent, this
+    approach is much faster than using scipy lstsq to fit the
+    data.
+
+    Args:
+        x:
+            ndarray, x axis data
+        y:
+            ndarray, y axis data
+        axis:
+            Optional argument, specifies the axis to operate on.
+        gradient_only:
+            Boolean, if true only returns the gradient.
+
+    Returns:
+        The gradient and the y-intercept in 2 1d numpy arrays.
+    """
+    if not isinstance(axis, tuple):
+        axis = (axis,)
+
+    # Check that there are no spurious NaNs in one of the arrays
+    # only - this will mess up the mean.
+    if not (np.isnan(y) == np.isnan(x)).all:
+        raise ValueError("Positions of NaNs do not match in x and y")
+
+    # Finds a compatible shape for the means to be reshaped into
+    if not x.shape == y.shape:
+        raise ValueError("Shape of x and y do not match")
+    shape = list(x.shape)
+    for ax in axis:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable name "ax" doesn't conform to snake_case naming style](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5319684)